-Werror makes it difficult for distro maintainers to build with newer GCC

I see that `generate/unix/Makefile.config` has `-Werror` in it. `-Werror` is a great development tool. But it breaks builds with newer versions of GCC. To help distro maintainers, I recommend you make it optional. Otherwise we have to patch it away. Or we can never update GCC. Thank you.

For instance, with GCC 12.2.0, we get:

```
../../../source/components/utilities/utdebug.c: In function ‘AcpiUtInitStackPtrTrace’:
../../../source/components/utilities/utdebug.c:188:31: error: storing the address of local variable ‘CurrentSp’ in ‘AcpiGbl_EntryStackPointer’ [-Werror=dangling-pointer=]
  188 |     AcpiGbl_EntryStackPointer = &CurrentSp;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
../../../source/components/utilities/utdebug.c:185:29: note: ‘CurrentSp’ declared here
  185 |     ACPI_SIZE               CurrentSp;
      |                             ^~~~~~~~~
In file included from ../../../source/include/acpi.h:173,
                 from ../../../source/components/utilities/utdebug.c:154:
../../../source/include/acglobal.h:334:41: note: ‘AcpiGbl_EntryStackPointer’ declared here
  334 | ACPI_GLOBAL (ACPI_SIZE *,               AcpiGbl_EntryStackPointer);
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
../../../source/include/acpixf.h:188:17: note: in definition of macro ‘ACPI_GLOBAL’
  188 |     extern type name
```

I think we have fixes for these already, they will appear in the next release of ACPICA.


From: Valentin David ***@***.***>
Sent: Wednesday, September 28, 2022 12:24 PM
To: acpica/acpica ***@***.***>
Cc: Moore, Robert ***@***.***>; Comment ***@***.***>
Subject: Re: [acpica/acpica] -Werror makes it difficult for distro maintainers to build with newer GCC (Issue #798)


For instance, with GCC 12.2.0, we get:

../../../source/components/utilities/utdebug.c: In function ‘AcpiUtInitStackPtrTrace’:



../../../source/components/utilities/utdebug.c:188:31: error: storing the address of local variable ‘CurrentSp’ in ‘AcpiGbl_EntryStackPointer’ [-Werror=dangling-pointer=]



  188 |     AcpiGbl_EntryStackPointer = &CurrentSp;



      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~



../../../source/components/utilities/utdebug.c:185:29: note: ‘CurrentSp’ declared here



  185 |     ACPI_SIZE               CurrentSp;



      |                             ^~~~~~~~~



In file included from ../../../source/include/acpi.h:173,



                 from ../../../source/components/utilities/utdebug.c:154:



../../../source/include/acglobal.h:334:41: note: ‘AcpiGbl_EntryStackPointer’ declared here



  334 | ACPI_GLOBAL (ACPI_SIZE *,               AcpiGbl_EntryStackPointer);



      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~



../../../source/include/acpixf.h:188:17: note: in definition of macro ‘ACPI_GLOBAL’



  188 |     extern type name



—
Reply to this email directly, view it on GitHub<https://github.com/acpica/acpica/issues/798#issuecomment-1261368713>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAHL7LCYWAUMZ66YFAIS4ADWASLOHANCNFSM6AAAAAAQXDHRLM>.
You are receiving this because you commented.Message ID: ***@***.******@***.***>>

--- a/generate/efi/Makefile.config
+++ b/generate/efi/Makefile.config
@@ -150,7 +150,6 @@ CWARNINGFLAGS = \
 	-Wall\
 	-Wbad-function-cast\
 	-Wdeclaration-after-statement\
-	-Werror\
 	-Wformat=2\
 	-Wmissing-declarations\
 	-Wmissing-prototypes\
@@ -162,6 +161,11 @@ CWARNINGFLAGS = \
 	-Wmissing-parameter-type\
 	-Wold-style-declaration\
 	-Wtype-limits
+
+ifneq ($(NOWERROR),TRUE)
+CWARNINGFLAGS += -Werror
+endif
+
 #
 # Extra warning flags (for possible future use)
 #

--- a/generate/unix/Makefile.config
+++ b/generate/unix/Makefile.config
@@ -203,7 +203,6 @@ CWARNINGFLAGS = \
     -Wall\
     -Wbad-function-cast\
     -Wdeclaration-after-statement\
-    -Werror\
     -Wformat=2\
     -Wmissing-declarations\
     -Wmissing-prototypes\
@@ -212,6 +211,10 @@ CWARNINGFLAGS = \
     -Wswitch-default\
     -Wpointer-arith\
     -Wundef
+
+ifneq ($(NOWERROR),TRUE)
+CWARNINGFLAGS += -Werror
+endif
 
 #
 # Common gcc 4+ warning flags

--- a/generate/unix/iasl/Makefile
+++ b/generate/unix/iasl/Makefile
@@ -290,6 +290,12 @@ CFLAGS += \
     -I$(ASL_COMPILER)\
     -I$(OBJDIR)
 
+ifeq ($(NOWERROR),TRUE)
+WERROR_FLAGS=
+else
+WERROR_FLAGS=-Werror
+endif
+
 #
 # Common Rules
 #
@@ -360,32 +366,32 @@ $(OBJDIR)/prparserparse.c $(OBJDIR)/prparser.y.h :       $(ASL_COMPILER)/prparse
 #
 $(OBJDIR)/aslcompilerlex.o :   $(OBJDIR)/aslcompilerlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/aslcompilerparse.o : $(OBJDIR)/aslcompilerparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/dtcompilerparserlex.o :      $(OBJDIR)/dtcompilerparserlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/dtcompilerparserparse.o :    $(OBJDIR)/dtcompilerparserparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/dtparserlex.o :      $(OBJDIR)/dtparserlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/dtparserparse.o :    $(OBJDIR)/dtparserparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/prparserlex.o :      $(OBJDIR)/prparserlex.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<
 
 $(OBJDIR)/prparserparse.o :    $(OBJDIR)/prparserparse.c
 	@echo "- " "Intermediate" $<
-	@$(CC) -c $(CFLAGS) -Wall -Werror -o$@ $<
+	@$(CC) -c $(CFLAGS) -Wall $(WERROR_FLAGS) -o$@ $<

